src/Model/Payment/AbstractPayment.php

```diff
@@ -120,10 +127,50 @@ public function isModuleActive(): bool
      * Checks if a payment method is enabled
      *
      * @return bool
+     * @throws ShopgateLibraryException
      */
     public function isEnabled(): bool
     {
-        return (bool) $this->scopeConfig->getConfigByPath(static::XML_CONFIG_ENABLED)->getValue();
+        $this->validatePaymentSettings();
+
+        return (bool)$this->scopeConfig->getConfigByPath(static::XML_CONFIG_ENABLED)->getValue();
+    }
+
+    /**
+     * Validate the payment settings
+     *
+     * @throws ShopgateLibraryException
+     */
+    private function validatePaymentSettings(): void
+    {
+        if (static::XML_CONFIG_ENABLED === '') {
+            $this->throwValidateException('No XML config enabled path set for payment');
+        }
+
+        if (static::XML_CONFIG_ORDER_STATUS === '') {
+            $this->throwValidateException('No XML config order status set for payment');
+        }
+
+        if (static::MODULE_NAME === '') {
+            $this->throwValidateException('No module name set for payment');
+        }
+
+        if (static::PAYMENT_CODE === '') {
+            $this->throwValidateException('No payment code set for payment');
+        }
+    }
```

Comment:
this is not quite what I meant, you need to throw an error in every function that uses the constant, so the enabled uses `static::XML_CONFIG_ENABLED`, that's where u need 1 if statement.
2. the isModuleActive uses `static::MODULE_NAME`, that's where u need the if statement for that
3. etc...

Reply:
pls. check